### PR TITLE
The Cascading Crusade or: How I Learned to Stop Worrying and Love Super-Duper-Fiesta

### PR DESCRIPTION
**Pull request by sardred**
# TL;DR: _MISC STYLING INTENSIFIES_
Lining things up, standardizing things, reducing redundant redundancy. 

_this is why i dont get invited to parties_
## Changelog
- Added a Card component, complete with styling and everything, for somewhat obvious reasons
  - A lot of "duplicate" styling has been removed thanks to this.
  - Dialog, IssueStatus, Issue, ConcludedIssue now all use Card. Might've forgotten another one.
- Converted _everything_ to rem, and more specifically changed a rem from 10px to default 16px
- Added some basic styling for Issueform and Users
  - Using checkboxes for voting rights on Users, but since we can't style them we're probably better off changing them for a custom flaticon thing in the future.
- Added [normalize.css](https://necolas.github.io/normalize.css/)
- Various tweaks here and there to tag-use and styling
  - Perhaps most notably is the use of h1 for the header, h2 for card titles, h3 for subtitles and h4 for form "labels". 
  - I went with a main div and styling directly targetting it because "The content inside the <main> element should be unique to the document." and "There must not be more than one <main> element in a document.", so there should be no misfire.
- Removed Roboto
  - **Why?** Prokom has been critical towards the use of Roboto for Online earlier (read: [regme](https://github.com/dotKom/regme)), and personally I thought it looked sort of cramped at that font-weight anyways.

## Y'all like images?
Have some before-and-afters - my treat.

### End-user index
index0
-
Note: The IssueStatus _does_ fill the row in this PR, just fixed it after the screenshot and was kinda lazy.
index1

### Admin index
admin0
-
admin1

### Issue form
form0
-
form1

### Edit dialog in issue form
edit0
-
edit1

### Registration dialog
reg0
-
reg1

### User admin
users0
-
users1

## What did we learn from this PR?
- for the love of god, just make a list of what you're doing, try your very best to avoid bikeyakshedshaving, make smaller PRs
- commit more often (but not necessarily smaller commits)
  - In the same vein, be more clear on what is being committed. While CSS often boils down to "tweaking" and the likes, try to keep commits structured
- getting some sleep and coming back to the work later helps _a lot_. suddenly the task at hand was trivial
- air your thoughts, be it towards fellow team members, a rubber duck or slackbot
- stop writing long pr texts that don't matter towards the actual code for the sake of half-assed comedic value

### CODE REVIEW

**Comment by sardred** [comment]
@sklirg: Checkbox does feel excessively simple, I agree. I'm thinking it can be replaced with a two-button toggle where each button is a flaticon, use some red and green and grays and you got yourself an intuitive interface. Also makes more sense to get a warning when clicking a button than a checkbox, like you said.

**Comment by sklirg** [comment]
Purrr-fect. 👍

File: client/src/components/App/index.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React, { PropTypes } from 'react';
 import { connect } from 'react-redux';
+import 'normalize.css';
```
Comment (duvholt) [inline client/src/components/App/index.js]: I think it makes more sense to include this in `src/index.js`.

File: client/src/css/Alternative.css [hunk 2]
```diff
@@ -1,43 +1,45 @@
...
-  background: #6b71a5;
-  color: #ffffff;
-  font-size: 1.8rem;
+  margin: 0.5rem 0;
+  background-color: #6B71A5;
+  color: #FFF;
   display: flex;
-  box-shadow: 0px -2px 0px 0px rgba(0, 0, 0, 0.25) inset,
-              0px 2px 1px 0px rgba(0, 0, 0, 0.2);
+  box-shadow: 0 -0.125rem 0 0 rgba(0, 0, 0, 0.25) inset,
```
Comment (duvholt) [inline client/src/css/Alternative.css]: Here we actually want pixels.
I've done the exact same thing before and came to the conclusion that nobody will remember how much 0.125 actually is.

File: client/src/css/Alternative.css [hunk 3]
```diff
@@ -1,43 +1,45 @@
...
+              0 0.125rem 0.0625rem 0 rgba(0, 0, 0, 0.2);
 }
 
 .Alternative:last-of-type {
   margin-bottom: 0;
 }
 
 .Alternative:hover {
-  background: #8d93c7;
+  background-color: #8D93C7;
```
Comment (duvholt) [inline client/src/css/Alternative.css]: Uppercase hex:
I don't like this. The way I pick colors is using Chrome dev tools and this defaults to lowercase. 
Meaning that every time I want to change a hex color I have to manually rewrite it to uppercase or end up with a mix. 

background -> background-color:
I don't see any reason for this change, but I don't really care that much either.

File: client/src/css/Alternative.css [hunk 4]
```diff
@@ -1,43 +1,45 @@
...
 }
 
 .Alternative label {
   flex-grow: 1;
-  padding: 1rem 3rem;
+  padding: 1rem;
+}
+
+.Alternative:hover label {
+  cursor: pointer;
```
Comment (duvholt) [inline client/src/css/Alternative.css]: `cursor: pointer` should only be used for anchors. 

I made an exception for the header buttons for consistency.

File: client/src/css/App.css [hunk 5]
```diff
@@ -1,39 +1,40 @@
...
 
 .ActiveIssue-components > * {
-  margin: 4rem 1rem 0 1rem;
+  margin: 0 0.5rem 0 0.5rem;
 }
 
 .ActiveIssue-Vote-wrapper {
   flex-grow: 4;
-  flex-basis: 30rem;
+  flex-basis: 20rem;
```
Comment (duvholt) [inline client/src/css/App.css]: If I remember correctly this was done because some questions can have a lot of text.
Not entirely sure though.

File: client/src/css/Button.css [hunk 6]
```diff
@@ -1,50 +1,57 @@
 .Button {
-  margin: 2rem 0 0 0;
-  padding: 1rem;
-  font-size: 1.1em;
-  border: 1px solid transparent;
-  background: transparent;
-  color: #636bb2;
+  margin: 1rem 0 0 0;
+  border: 0.0625rem solid transparent;
```
Comment (duvholt) [inline client/src/css/Button.css]: 1px

File: client/src/css/Button.css [hunk 7]
```diff
@@ -1,50 +1,57 @@
...
-  color: #ffffff;
-  background: #6d7194;
+  color: #FFF;
+  background-color: #6D7194;
 }
 
 .Button:disabled {
-  background: #71727d;
+  background-color: #71727D;
+  opacity: 0.5;
```
Comment (duvholt) [inline client/src/css/Button.css]: Wouldn't it be better to change the background color than change opacity?

File: client/src/css/IssueForm.css [hunk 8]
```diff
@@ -18,32 +16,88 @@
...
-  padding: 0.8rem;
-  max-width: 50rem;
+  border: 0.0625rem solid #D0D0D0;
+  background-color: #F5F5F5;
+  max-width: 32rem;
 }
 
 .IssueForm textarea {
-  height: 20rem;
+  -webkit-box-sizing: border-box;
```
Comment (duvholt) [inline client/src/css/IssueForm.css]: box-sizing is so well supported that there is no need for browser prefixes.

File: client/src/components/Admin/Issue.js [hunk 9]
```diff
@@ -1,17 +1,20 @@
...
 const Issue = ({ closeIssue, issue }) => (
-  <div className="IssueAdmin">
-    <div className="IssueAdmin-heading">
+  <Card
+    classes="IssueAdmin"
+  >
+    <div>
       <Pin code="DEADBEEF" />
-      <h2 className="IssueAdmin-title">Aktiv sak</h2>
+      <p className="IssueAdmin-title">Aktiv sak</p>
```
Comment (duvholt) [inline client/src/components/Admin/Issue.js]: Isn't this still a header?

File: client/src/components/Admin/IssueForm/index.js [hunk 10]
```diff
@@ -101,13 +101,19 @@ class IssueForm extends React.Component {
           hidden={!showActiveIssueWarning}
         >Det er allerede en aktiv sak!</p>
         <label className="IssueForm-textarea">
-          <div className="IssueForm-label">Beskrivelse</div>
+          <h4 className="IssueForm-label">Beskrivelse av saken</h4>
```
Comment (duvholt) [inline client/src/components/Admin/IssueForm/index.js]: 4 seems high. I don't have the DOM in front of me, but I think this should be h2 maybe h3?

File: client/src/css/Alternative.css (hunk 2 above)
Comment (sardred) [inline client/src/css/Alternative.css]: Fine by me, just seems sort of odd to make an exception on this one case just because people can't be bothered. Will fix regardless, I do find it easier to just write 1-2px, but if you're using rem for whatever benefits it hold, why make exceptions, you know?

File: client/src/css/Alternative.css (hunk 3 above)
Comment (sardred) [inline client/src/css/Alternative.css]: I was going off some article that I skimmed through last night, it all made sense at the time, but seeing as I can't find it anymore I'm just assuming I had a hallucination or something. [This](https://github.com/mdo/css-perf#background-vs-background-color) actually indicates that background is the preferred method anyways.

Will switch to lowercase, doesn't matter at all to me, just made an arbitrary choice at the time. PyCharm will convert all of them in an instant.

File: client/src/css/Alternative.css (hunk 4 above)
Comment (sardred) [inline client/src/css/Alternative.css]: I don't mind changing it, but can you elaborate on why that is?

File: client/src/css/App.css (hunk 5 above)
Comment (sardred) [inline client/src/css/App.css]: The flex-basis is actually larger due to 1rem now being 16px, meaning it went from 300px to 320.

Comment (duvholt) [inline client/src/css/App.css]: Forgot that :fearful:

File: client/src/css/Button.css (hunk 7 above)
Comment (sardred) [inline client/src/css/Button.css]: Personally I feel opacity is underutilized. We have a white background, or at least somewhat white throughout the entire page. Reducing opacity is equivalent to lightening the background color, which is sort of what we want. 
It also stems from my aversion to throwing out what are essentially the magical numbers of hex-codes. It feels like we have countless shades of indigo, changing opacity very clearly defines that this is that one shade, just 50% brighter.

File: client/src/components/Admin/Issue.js (hunk 9 above)
Comment (sardred) [inline client/src/components/Admin/Issue.js]: That one was sort of a pickle. I couldn't elegantly fit the pin and "title" into the card subtitle, so I just let them be card content. Going off of the style that was there previously it didn't seem to be styled particularly prominently, so I mostly let it be. h2 is sort of big regardless, h4 could be used here potentially.

File: client/src/components/Admin/IssueForm/index.js (hunk 10 above)
Comment (sardred) [inline client/src/components/Admin/IssueForm/index.js]: I could stretch up to 3 if you feel 4 looks too much like standard text, but I really do think h4 is best suited here. h2 is definitely too large, considering you have one of those on every fieldset of the form

File: client/src/css/Alternative.css (hunk 4 above)
Comment (duvholt) [inline client/src/css/Alternative.css]: https://www.w3.org/wiki/CSS/Properties/cursor

On the web it's pretty chaotic, but desktops generally don't use pointer for things like buttons etc.
Just try clicking around in chrome settings for example. 
I don't think there is any real consensus on this though. 
 
IMO it's better to not use pointer as a crutch to indicate that elements are clickable. Mobile devices won't display any pointers for example.

File: client/src/components/Admin/IssueForm/index.js (hunk 10 above)
Comment (duvholt) [inline client/src/components/Admin/IssueForm/index.js]: No, I don't mean the size. 

h1, h2 etc have a semantic meaning and tools that check for accessibility will cry about using them out of order. 
If the size is too small/big just set font-size. It's better than relying on the default browser size imo.

http://khan.github.io/tota11y/ is nice for finding some simple things like low contrast and misuse of h1, h2.

File: client/src/css/Button.css (hunk 7 above)
Comment (duvholt) [inline client/src/css/Button.css]: In some cases I like to use rgba, but I don't think it makes sense in this case.
There is no guarantee that the button will always be used on a white background and therefore you might end up with some odd looking background colors. 

Magic hex colors is a problem yeah, but that's unfortunately how CSS works. Less/Sass variables and methods like `lighten($color, 0.5)` are pretty neat tho. I don't think it's worth introducing Less or Sass at this point though.

File: client/src/css/Alternative.css (hunk 2 above)
Comment (duvholt) [inline client/src/css/Alternative.css]: I don't have any sources as to why this is "better", but from what I've noticed the browser will round border-width to nearest pixel anyway. 

"Everyone" knows what a 1px thick border looks like, but fiddling with tiny numbers like 0.0625 is just annoying. Sure you could round it, but what do you do when you want the border to be just a bit thicker? Then you have to know (or convert from pixels) to find out the nearest rem that will provide you with that. Overall it's just too much hassle imo.

**Comment by myth** [comment]
Yes, I would like one neatly styled interface for my hobby dashboard please.

Thank you.

- Bob
